Add automated deployment scripts

whereisaaron opened the pull request:
Thank you for sharing this container build, it was just what I needed. 👍 

I wanted to automate the deployment and removal. I forked and added deployment/removal scripts and arranged for everything configurable to work off environment variables. The environment variables are documented in the updated README. The scripts are:
- Deploy.sh
- Display.sh
- Delete.sh

You may not want any of this, but I wanted to offer it back anyway. 😃

whereisaaron committed:
Add deployment scripts to create Secret, ConfigMap, and DaemonSet

whereisaaron committed:
Update README to describe deployment scripts and acknowledge the fork

whereisaaron committed:
Tweak deployment and add a Delete.sh script

whereisaaron committed:
Add SetVersion.sh and set image tag to a 0.1.1

whereisaaron committed:
Switch to versions >1 because forked project is already >1

whereisaaron committed:
Prepare README and image name for offering back upstream as a PR

whereisaaron committed:
Add 'Removal' section to the README

whereisaaron committed:
Make the CloudWatch log group configurable too, update README

eljasala commented:
Thank you for your contribution and sorry for a bit of delay getting back to you. Would you mind signing the CLA at https://callstats.io/icla.pdf?

eljasala commented on Delete.sh:
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
```
Do you mind changing the filenames to lowercase for consistency across our other projects?

whereisaaron replied:
Heh, they're capitalized for consistency with *our* other projects 😄 
No problem, I can change them on this branch.

whereisaaron replied:
Changed.

eljasala commented on README.md:
```diff
@@ -1,17 +1,34 @@
-kubectl apply -f fluentd-cloudwatch-daemonset.yaml
+export AWS_ACCESS_KEY_ID=<your key>
+export AWS_SECRET_ACCESS_KEY=<your secret>
+export AWS_DEFAULT_REGION=<your region>
```
This is a bit risky as there's unfortunately a bit high chance that the user could just skip this step and end up using their currently loaded credentials (which likely have quite broad permissions). It would be better to name the variables differently from AWS's default credentials and also possibly suggest passing the variables directly to the deploy script, ex.
```
LOGGING_AWS_ACCESS_KEY_ID=foobar LOGGING_AWS_SECRET_ACCESS_KEY=secretfoo  LOGGING_AWS_DEFAULT_REGION=bar ./deploy.sh 
```
This would prevent the variables from be exposed to other processes that are launched later, would not overwrite the currently loaded credentials and generally user only needs to run the script once.

whereisaaron replied:
Will do

whereisaaron replied:
Changed and updated README

eljasala reviewed (changes requested):
Overall design LGTM, but I would prefer to see couple of changes.

whereisaaron committed:
Lower case all the deployment scripts
Change AWS environment variables to have a 'LOGGING_' prefix
Update README and suggest setting AWS environment variables on deploy.sh command line

whereisaaron committed:
Add a ./ to example in README

whereisaaron committed:
Lower case one more 'D' in README

whereisaaron commented:
@eljasala  I completed the requested changes.

eljasala reviewed (approved):
Thanks for the changes, LGTM. :)

Could you sign the CLA? I can merge the things afterwards.

whereisaaron commented:
@eljasala oh right, a manual PDF one 😃 , will do, I'll fill it in and email.

whereisaaron commented:
@eljasala sorry for the delay. I just emailed through the signed CLA.

eljasala commented:
Thanks, got it.

And thanks for the contribution! :+1: They should make things easier for people to deploy it.
